File: src/Panorama.Web.Host/src/shared/service-proxies/scenography/dtos/UpdateSceneDto.ts
import {IEntityDto} from "@shared/service-proxies/common/dtos/EntityDto";

export interface IUpdateSceneDto extends IEntityDto {
    name: string;
    description: string | undefined;
}

export class UpdateSceneDto implements IUpdateSceneDto {
    id: number = 0;
    name: string = undefined;
    description: string | undefined = undefined;
}

File: src/Scenography/teatro.shared/Teatro.Shared/Scenes/Dtos/UpdateSceneDto.cs
using System.ComponentModel.DataAnnotations;
using Teatro.Shared.Bases.Dtos;

namespace Teatro.Shared.Scenes.Dtos;

// TODO: do not expose db id's
public class UpdateSceneDto : EntityDto<long>
{
    [Required(AllowEmptyStrings = false)]
    [MaxLength(SceneConstants.MaxNameLength)]
    public string Name { get; set; }
    
    [MaxLength(SceneConstants.MaxDescriptionLength)]
    public string Description { get; set; }
}